### DISCUSSION TITLE
Setting variables based OS architecture

### DISCUSSION THREAD

**Discussion by arky** [community/community#25949] [Actions]
<p>Am trying to set variable <em>architecture: x86</em> for setup-python if the <em>matrix.config.os.arch == ‘ia32’</em> But I get an syntax error. What is correct way to set variables here?</p>
<pre><code class="lang-auto"> strategy:
      matrix:
        config:
          - { os: macos-latest, arch: x64, name: macos-latest }
          - { os: ubuntu-latest, arch: x64, name: ubuntu-latest }
          - { os: windows-latest, arch: x64, name: windows-latest }
          - { os: windows-latest, arch: ia32, name: windows-latest-32 }

    steps:
      - uses: actions/checkout@v1
      - name: Set up Python 3.8
        uses: actions/setup-python@v1
        with:
          python-version: 3.8
          if: ${{ matrix.config.os.arch == 'ia32'}} 
          architecture: x86 
</code></pre>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/arky">@arky</a>,</p>
<p>According to the configurations of the matrix in your workflow, the ‘<strong>os</strong>’ and the ‘<strong>arch</strong>’ are the same level options in the matrix. The ‘<strong>arch</strong>’ is not a child option of the ‘<strong>os</strong>’.</p>
<p>So, if you want to access the value of the ‘<strong>arch</strong>’ option, the correct expression syntax should be ‘<strong>matrix.config.arch</strong>’, not ‘<strong>matrix.config.os.arch</strong>’.</p>
<p>Similarly, using ‘<strong>matrix.config.os</strong>’ to access the value of the ‘<strong>os</strong>’ option, and ‘<strong>matrix.config.name</strong>’ to access the ‘<strong>name</strong>’.<br>
Below is an example as reference:<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/ForksForTest/TestMatrix/actions/runs/172450304/workflow" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/46307996/181094275-e1183262-fc6c-410b-88e1-29cff2b11965.png" class="thumbnail onebox-avatar" width="420" height="420">

<h3><a href="https://github.com/ForksForTest/TestMatrix/actions/runs/172450304/workflow" target="_blank" rel="nofollow noopener">ForksForTest/TestMatrix</a></h3>

<p>Contribute to ForksForTest/TestMatrix development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by arky** [comment]
<p><a class="mention" href="/brightran">@brightran</a> Thanks again! I missed that. Looks like I am hitting multiple errors here. Firstly I can’t use if here.  Wonder if there is shorthand to return ‘x86’ string if ‘matrix.config.arch == ia32’ is true.</p>
<p>architecture:  {{ if: true return string }}</p>
<p>The syntax error</p>
<pre><code class="lang-auto">
##[warning]Unexpected input(s) 'if', valid inputs are ['python-version', 'architecture']
Run actions/setup-python@v1
  with:
    python-version: 3.8
    if: false
    architecture: x86
##[error]Version 3.8 with arch x86 not found
Available versions:
</code></pre>
<p>Also python-setup doesn’t seem to have ‘x86’ architecture.</p>

**Comment by arky** [comment]
<p>Resolved. Thanks for your help <a class="mention" href="/brightran">@brightran</a>!</p>
<pre><code class="lang-auto">  matrix:
        config:
          - { os: macos-latest, arch: x64, python-arch: x64, name: macos-latest }
          - { os: ubuntu-latest, arch: x64, python-arch: x64, name: ubuntu-latest }
          - { os: windows-latest, arch: x64, python-arch: x64, name: windows-latest }
          - { os: windows-latest, arch: ia32, python-arch: x86, name: windows-latest-32 }

    steps:
      - uses: actions/checkout@v1
      - name: Set up Python 3.8
        uses: actions/setup-python@v1
        with:
          python-version: '3.8'
          architecture: ${{ matrix.config.python-arch }}
      ```</code></pre>
